hawk/lib/crm_script.rb

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2015 Kristoffer Gronlund <kgronlund@suse.com>
+# See COPYING for license information.
+
+require 'pty'
+require 'json'
+
+module CrmScript
+  def crmsh_escape(s)
+    s.to_s.gsub(/[&"'>< ]/) { |special| "\\#{special}" }
+  end
+  module_function :crmsh_escape
+
+  def run(jsondata)
+    user = 'hacluster'#'root'
+    old_home = ENV['HOME']
+    begin
+      ENV['HOME'] = begin
+                      require 'etc'
+                      Etc.getpwnam(user)['dir']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.